packages/components/src/components/ViewYAML/ViewYAML.js

```diff
@@ -15,29 +15,54 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import jsYaml from 'js-yaml';
 import classNames from 'classnames';
+import { Prism as SyntaxHighlight } from 'react-syntax-highlighter';
 
 const ViewYAML = props => {
-  const { className, dark, resource } = props;
-
-  return (
-    <div
-      className={classNames('bx--snippet--multi', className, {
-        'tkn--view-yaml--dark': dark
-      })}
-    >
-      <code>
-        <pre>{jsYaml.dump(resource)}</pre>
-      </code>
-    </div>
-  );
+  const { className, dark, resource, enableSyntaxHighlighting = false } = props;
+
+  const renderRaw = (clz, yaml) => {
+    return (
+      <div>
+        <code className={clz}>
+          <pre>{yaml}</pre>
+        </code>
+      </div>
+    );
+  };
+
+  const renderSyntaxHighlight = (clz, yaml) => {
+    return (
+      <SyntaxHighlight
+        className={clz}
+        language="yaml"
+        useInlineStyles={false}
+        codeTagProps={{}}
+      >
+        {yaml}
+      </SyntaxHighlight>
+    );
+  };
+
+  const clz = classNames('bx--snippet--multi', className, {
+    'tkn--view-yaml--dark': dark
+  });
+
+  const yaml = jsYaml.dump(resource);
+
+  if (enableSyntaxHighlighting) {
+    return renderSyntaxHighlight(clz, yaml);
+  } else {
+    return renderRaw(clz, yaml);
```

Comment:
test are failing due to a lint error here:

```
/home/prow/go/src/github.com/tektoncd/dashboard/packages/components/src/components/ViewYAML/ViewYAML.js
  54:10  error  Unnecessary 'else' after 'return'  no-else-return
```

Reply:
Thank you for your help. I won't forget to run lint next time. :)

Reply:
No worries, it can be easy to miss. Depending on the editor you're using it might be useful to install a linter plugin that's compatible with our eslint / prettier config so you get feedback while you're editing. I find these very helpful.

I'll take a look at our webpack config too as we should be catching lint error during the dev build.